src/screens/details.tsx

```diff
@@ -1,20 +1,69 @@
 import React from 'react';
-import { StyleSheet, View } from 'react-native';
-import { Text } from 'react-native-paper';
+import { StyleSheet, View, Linking } from 'react-native';
+import { Button, Text } from 'react-native-paper';
+import DropDown from '../components/dropdown';
+import { items } from '../trainData/avaliableStations.json';
+import { DropdownItem } from '../types/DropDownItem';
+import { ScreenNavigationProps } from '../routes';
+
 
 const styles = StyleSheet.create({
   container: {
     flex: 1,
-    backgroundColor: '#eee',
-    alignItems: 'center',
-    justifyContent: 'center',
+    backgroundColor: 'lightblue',
+    //alignItems: 'center',
+    //justifyContent: 'center',
   },
 });
 
-const DetailsScreen: React.FC = () => (
-  <View style={styles.container}>
-    <Text>Details Screen</Text>
-  </View>
-);
+const defaultStation: DropdownItem = {
+  displayName: 'Pick a station',
+  value: null,
+};
+
+type DetailsScreenProps = ScreenNavigationProps<'Details'>;
+
+const DetailsScreen: React.FC<DetailsScreenProps> = ({ navigation }) => {
+
+  const [departureStation, setDeparture] = React.useState(defaultStation);
+  const [arrivalStation, setArrival] = React.useState(defaultStation);
+
+  function openTrainTimesUrl(
```

Comment:
It's probably better to pull this out of the component into a helper file for tidiness and so that it can be used in multiple places.